File: dailyChallenge/javascript/2026/01/23.js
// Hex Validator

// Given a string, determine whether it is a valid CSS hex color. A valid CSS hex color must:

//     Start with a #, and
//     be followed by either 3 or 6 hexadecimal characters.

// Hexadecimal characters are numbers 0 through 9 and letters a through f (case-insensitive).


// 1. isValidHex("#123") should return true.
// 2. isValidHex("#123abc") should return true.
// 3. isValidHex("#ABCDEF") should return true.
// 4. isValidHex("#0a1B2c") should return true.
// 5. isValidHex("#12G") should return false.
// 6. isValidHex("#1234567") should return false.
// 7. isValidHex("#12 3") should return false.
// 8. isValidHex("fff") should return false.

function isValidHex(str) {
    if (str.charAt(0) !== "#") return false

    if (str.length !== 4 && str.length !== 7) return false

    const valid = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f']
    for (let ch of str.slice(1)) {   // skip the #
        if (!valid.includes(ch.toLowerCase())) return false
    }

    return true
}

console.log(isValidHex("#123"))
console.log(isValidHex("#123abc"))
console.log(isValidHex("#ABCDEF"))
console.log(isValidHex("#0a1B2c"))
console.log(isValidHex("#12G"))
console.log(isValidHex("#1234567"))
console.log(isValidHex("#12 3"))
console.log(isValidHex("fff"))

File: dailyChallenge/python/2026/01/23.py
# Hex Validator

# Given a string, determine whether it is a valid CSS hex color. A valid CSS hex color must:

#     Start with a #, and
#     be followed by either 3 or 6 hexadecimal characters.

# Hexadecimal characters are numbers 0 through 9 and letters a through f (case-insensitive).



# 1. is_valid_hex("#123") should return True.
# 2. is_valid_hex("#123abc") should return True.
# 3. is_valid_hex("#ABCDEF") should return True.
# 4. is_valid_hex("#0a1B2c") should return True.
# 5. is_valid_hex("#12G") should return False.
# 6. is_valid_hex("#1234567") should return False.
# 7. is_valid_hex("#12 3") should return False.
# 8. is_valid_hex("fff") should return False.

def is_valid_hex(s):
    if s[0] != '#':
        return False
        
    if ((len(s) - 1) % 3 != 0):
        return False
    
    for ch in s[1:]:
        if ch.lower() not in ['0','1','2','3','4','5','6','7','8','9','a','b','c','d','e','f']:
            return False
    return True


print(is_valid_hex("#123"))
print(is_valid_hex("#123abc"))
print(is_valid_hex("#ABCDEF"))
print(is_valid_hex("#0a1B2c"))
print(is_valid_hex("#12G"))
print(is_valid_hex("#1234567"))
print(is_valid_hex("#12 3"))
print(is_valid_hex("fff"))